piku/commands/deploy.py

```diff
@@ -1,40 +1,116 @@
+        drive = path.split(':')[0]
+        output = check_output(f'cmd /c vol {drive}:'.split()).decode()
+        label = output.split('\r\n')[0].split(' ').pop()
+        return expected_label in label.lower()
+    return expected_label in path.lower()
+
+
+def find_device_path(expected_label=DEFAULT_VOLUME_LABEL):
+    if platform.system() == 'Windows':
+        output = check_output('wmic logicaldisk where drivetype=2 get DeviceId , VolumeName'.split()).decode()
+        drives = [
+            line.split(' ')[0]
+            for line in output.split('\r\n')
+            if expected_label in line.lower()
+        ]
+        return drives[0] if drives else None
+    if platform.system() == 'Linux':
+        output = check_output('lsblk -l -o mountpoint,label,rm'.split()).decode()
+        lines = [line.split() for line in output.split('\n')]
+        drives = [
+            line[0]
+            for line in lines
+            if len(line) == 3 and expected_label in line[1].lower() and line[2] == '1'
+        ]
+        return drives[0] if drives else None
+    if platform.system() == 'Darwin':
+        output = check_output('mount'.split()).decode()
+        lines = [line.split() for line in output.split('\n')]
+        drives = [
+            line[2]
+            for line in lines
+            if len(line) > 4 and expected_label in line[2].lower()
+        ]
+        return drives[0] if drives else None
+    return None
+
+  def deploy(drive):
     print(f'Deploying project to device {drive}...')
     device.deploy(drive)
     print('Done')
 
 
 def deploy_command(args):
-    # get device
-    drive = args.device or device.find_device_path()
-
-    # check that we are in a piku project directory
+  # check that we are in a piku project directory
     if not config.valid():
         print('Refusing to deploy, unable to find piku project in current directory.')
         return
 
+  # get device
+    source = config.get('tool.piku.source')
+    expected_label = DEFAULT_VOLUME_LABEL
+    device = args.device or find_device_path(expected_label)
+    user_confirmed = False
+
+    drive = args.device or device.find_device_path()
+
     # check that we have a device found or specified
     if not drive:
         print('Unable find a device and deploy, please specify a device to deploy to.')
         return
 
     # check that device size and name are as expected to reduce chances of loading onto wrong device
-    if not device.has_correct_size(drive):
-        print('Refusing to deploy, specified CircuitPython drive is larger than expected (~2MB).')
-        return
-    if not device.has_correct_label(drive):
-        print('Refusing to deploy, expected device to have "circuitpy" in path.')
-        return
+    if not has_correct_size(device):
+        print('Specified CircuitPython drive is larger than expected (~{:0d}MB).'.format(int(DEFAULT_BOARD_CAPACITY // 1E6)))
+        if not warning_prompt(f'Are you sure you want to deploy to {device}?'):
```

Comment:
```suggestion
        if not args.device or not warning_prompt(f'Are you sure you want to deploy to {device}?'):
```
Only allow override only if the device was specified (not auto detected).  Otherwise someone might blast through the warning and erase their hard drive.